Implement EZP-24266: Indexable Relation field type

pspanja opened the pull request:
This PR resolves https://jira.ez.no/browse/EZP-24266

A subtask of https://jira.ez.no/browse/EZP-24232

This implements `Indexable` definition for `Relation` field type.

Since not all searchable field types have `Indexable` definition implemented, this required `RelationSearchBaseIntegrationTest`. This is temporary and will be removed when all searchable field types are covered.

pspanja commented:
This intentionally defines Content IDs in reverse, to demonstrate the difference in sorting caused by indexing the Content ID in Solr and Elasticsearch as string and having it as integer in Legacy search engine (`sort_key_int` column).

The question here is how do we handle this? Similar question exists with `ContentId` sort clause, as the Content ID as a property of Content document is also indexed as string. One option is indexing the ID as an integer value. As next storage engine might use hash string as IDs, that means schema/mappings will be dependent on the storage engine used. As order in the case of string IDs can't be predicted, does `ContentId` sort clause makes sense in the first place?

Feedback welcome.

andrerom commented:
+1 ( when Travis agrees :) )

pspanja commented:
@andrerom should I index it as integer then? I can also choose the IDs to mask the difference, but we should probably do something about it :)

andrerom commented:
Decision taken in talk:
- Use Id (ContentId) or string (Field value) in search engines for id's
-- Change tests to reflect that Legacy Engine uses int, and solr/es uses string, like already the case more or less for SortClause\ContentId tests.
- Add phpdoc on SortClause\ContentId and similar one on SortClause\Field making remark on this that order will depend on search engine.

pspanja commented:
Changed Elasticsearch mappings to index IDs a string and updated tests accordingly in https://github.com/ezsystems/ezpublish-kernel/commit/c3756ee35d1ba539f153dd8b559376fba23ed5c6.
Difference in sort per search engine is documented in https://github.com/ezsystems/ezpublish-kernel/commit/9448048716743a03475053fee922ed9c8010aae2.

This is now ready for review.

andrerom commented:
+1
